scripts/GasEstimation.ts

```diff
@@ -0,0 +1,420 @@
+  })) as SmartWallet;
+
+  const baseRelayRequest = {
+    request: {
+      relayHub: relayHub.address,
+      to: constants.AddressZero,
+      data: '0x',
+      from: owner.address,
+      nonce: (await smartWallet.nonce()).toString(),
+      value: '0',
+      gas: '0',
+      tokenContract: token.address,
+      tokenAmount: '0',
+      tokenGas: '0',
+      validUntilTime: '0',
+    },
+    relayData: {
+      gasPrice: GAS_PRICE,
+      feesReceiver: relayWorker.address,
+      callForwarder: smartWallet.address,
+      callVerifier: verifier.address,
+    },
+  };
+
+  await token.mint(TOKEN_AMOUNT_TO_TRANSFER + '00', smartWallet.address);
+
+  const smartWalletInitialBalance = await token.balanceOf(smartWallet.address);
+  const relayWorkerInitialBalance = await token.balanceOf(relayWorker.address);
+
+  const transferReceiver = ethers.Wallet.createRandom();
+
+  const completeReq: RelayRequest = await completeRelayRequest(
+    transferReceiver.address,
+    TOKEN_AMOUNT_TO_TRANSFER,
+    fees,
+    token,
+    baseRelayRequest,
+    smartWallet
+  );
+
+  const { signature } = await getSuffixDataAndSignature(
+    smartWallet,
+    completeReq,
+    owner
+  );
+
+  const relayCallResult = await relayHub
+    .connect(relayWorker)
+    .relayCall(completeReq, signature, {
+      gasLimit: GAS_LIMIT,
+      gasPrice: GAS_PRICE,
+    });
+  const txReceiptWithRelay = await relayCallResult.wait();
+
+  const smartWalletFinalBalance = await token.balanceOf(smartWallet.address);
+  const relayWorkerFinalBalance = await token.balanceOf(relayWorker.address);
+
+  assertRelayedTransaction(
+    smartWalletInitialBalance,
+    relayWorkerInitialBalance,
+    smartWalletFinalBalance,
+    relayWorkerFinalBalance,
+    BigNumber.from(TOKEN_AMOUNT_TO_TRANSFER),
+    BigNumber.from(fees)
+  );
+
+  const txReceiptWithoutRelay = await getTransferEstimationWithoutRelay(token);
+
+  printRelayGasAnalysis(txReceiptWithRelay, txReceiptWithoutRelay);
+}
+
+async function estimateDeployCost(tokenAmount = '0') {
+  const SMART_WALLET_INDEX = '1';
+  const { relayHub, owner, smartWalletFactory, token, verifier } =
+    await deployAndSetup();
+
+  let tokenGas = '0';
+
+  if (tokenAmount !== '0') {
+    const swAddress = await smartWalletFactory.getSmartWalletAddress(
+      owner.address,
+      constants.AddressZero,
+      SMART_WALLET_INDEX
+    );
+    await token.mint(tokenAmount, swAddress);
+
+    const dataForTransfer = token.interface.encodeFunctionData('transfer', [
+      relayWorker.address,
+      tokenAmount,
+    ]);
+
+    tokenGas = (
+      await getEstimatedGasWithCorrection(
+        swAddress,
+        token.address,
+        dataForTransfer
+      )
+    ).toString();
+  }
+
+  const deployRequest = {
+    request: {
+      data: '0x00',
+      from: owner.address,
+      nonce: '0',
+      relayHub: relayHub.address,
+      to: constants.AddressZero,
+      tokenAmount,
+      tokenContract: constants.AddressZero,
```

Comment:
```suggestion
     tokenContract: token.address,
```
This is what we discussed with Raul and needs to be fixed.